From 26099f44119c0f8c1b5eef9ba2ba6fb0be1776e5 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 5 Jul 2024 14:47:02 +0200
Subject: [PATCH 1016/1695] drivers: stm32_remoteproc: Add the support of the
 stm32mp21 series

Add CFG_STM32MP21 to support the stm32mp21 series.
The comment on the #endif is removed it exceeds the 80-character limit

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e1c82a7e056cab38f5fd02d21311fd445c213122 drivers: stm32_remoteproc: Add the support of the stm32mp21 series]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Ica6878d71dd07f4f5a6497a27d4ef0a7dc851787
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391730
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/remoteproc/stm32_remoteproc.c | 26 +++++++++++-----------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index 5b5104c16..77510cd08 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -16,9 +16,9 @@
 #include <libfdt.h>
 #include <mm/core_memprot.h>
 #include <mm/core_mmu.h>
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 #include <stm32_sysconf.h>
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif
 
 #define TIMEOUT_US_1MS	U(1000)
 
@@ -104,7 +104,7 @@ void *stm32_rproc_get(uint32_t rproc_id)
 	return rproc;
 }
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 /* Re-apply default access right on the memory regions */
 static TEE_Result
 stm32_rproc_release_mems_access(struct stm32_rproc_instance *rproc)
@@ -204,7 +204,7 @@ static TEE_Result stm32mp2_rproc_start(struct stm32_rproc_instance *rproc)
 					   mems[i].size))
 			continue;
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 		if (rproc->tzen) {
 			stm32mp_syscfg_write(A35SSC_M33_INITSVTOR_CR,
 					     rproc->boot_addr, INITVTOR_MASK);
@@ -218,7 +218,7 @@ static TEE_Result stm32mp2_rproc_start(struct stm32_rproc_instance *rproc)
 		}
 		boot_addr_valid = true;
 		break;
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif
 	}
 
 	if (!boot_addr_valid) {
@@ -270,11 +270,11 @@ static TEE_Result rproc_stop(struct stm32_rproc_instance *rproc)
 	if (res)
 		return res;
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 	/* Disable the TrustZone */
 	stm32mp_syscfg_write(A35SSC_M33_TZEN_CR, 0,
 			     A35SSC_M33_TZEN_CR_CFG_SECEXT);
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif
 
 	rproc->boot_addr = 0;
 	rproc->tzen = false;
@@ -630,7 +630,7 @@ static void stm32_rproc_cleanup(struct stm32_rproc_instance *rproc)
 
 static void stm32_rproc_a35ss_cfg(struct stm32_rproc_instance *rproc __unused)
 {
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 	stm32mp_syscfg_write(A35SSC_M33CFG_ACCESS_CR, rproc->m33_cr_right,
 			     A35SSC_M33_TZEN_CR_M33CFG_SEC |
 			     A35SSC_M33_TZEN_CR_M33CFG_PRIV);
@@ -638,7 +638,7 @@ static void stm32_rproc_a35ss_cfg(struct stm32_rproc_instance *rproc __unused)
 	stm32mp_syscfg_write(A35SSC_M33_TZEN_CR, 0,
 			     A35SSC_M33_TZEN_CR_CFG_SECEXT);
 
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif
 }
 
 static TEE_Result stm32mp25_rproc_pm(enum pm_op op, unsigned int pm_hint,
@@ -663,10 +663,10 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 	if (!rproc)
 		return TEE_ERROR_OUT_OF_MEMORY;
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 	rproc->m33_cr_right = A35SSC_M33_TZEN_CR_M33CFG_SEC |
 			      A35SSC_M33_TZEN_CR_M33CFG_PRIV;
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif
 
 	rproc->cdata = comp_data;
 	rproc->fdt = fdt;
@@ -690,7 +690,7 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 	if (res)
 		goto err;
 
-#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23)
+#if defined(CFG_STM32MP25) || defined(CFG_STM32MP23) || defined(CFG_STM32MP21)
 	if (rproc->cdata->ns_loading) {
 		/*
 		 * The remote firmware will be loaded by the non secure
@@ -700,7 +700,7 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 		rproc->m33_cr_right = A35SSC_M33_TZEN_CR_M33CFG_PRIV;
 	}
 	stm32_rproc_a35ss_cfg(rproc);
-#endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
+#endif
 
 	if (!rproc->cdata->ns_loading)
 		SLIST_INSERT_HEAD(&rproc_list, rproc, link);
-- 
2.47.2

